app/services/course/assessment/question/programming/cpp/cpp_package_service.rb

```diff
@@ -0,0 +1,248 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::Programming::Cpp::CppPackageService < \
```

Comment:
@zhengwei143 This service is a little long, would it be possible to break this down into smaller, independent pieces - any suggestions? 

Don't constrain yourself to the methods defined in `Course::Assessment::Question::Programming::LanguagePackageService`, as that can be refactored in the controller later on.